src/frontend/hooks/server/projects.ts

```diff
@@ -78,3 +79,21 @@ export function useProjectSettings() {
     },
   });
 }
+
+export function useLeaveProject() {
+  const mapeoApi = useApi();
+  const projectId = usePersistedProjectId(store => store.projectId);
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationFn: () => {
+      if (!projectId) throw new Error('project Id does not exist');
+      return mapeoApi.leaveProject(projectId);
+    },
```

Comment:
thoughts on making this hook more agnostic about which project is being left, and letting the caller of the mutation provide the project id? 

```suggestion
    mutationFn: ({ projectId }: { projectId: string }) => {
      return mapeoApi.leaveProject(projectId);
    },
```

Reply:
https://github.com/digidem/comapeo-mobile/pull/526/commits/7a1751dcc4699ca9f73ec3357ef61b8364812bc5